Tools/Scripts/libraries/webkitscmpy/webkitscmpy/program/pull_request.py

```diff
@@ -354,16 +357,47 @@ def create_pull_request(cls, repository, args, branch_point, callback=None, unbl
 
         existing_pr = None
         if remote_repo.pull_requests:
+            user, _ = remote_repo.credentials(required=False)
+
             log.info("Checking if PR already exists...")
             existing_pr = cls.find_existing_pull_request(repository, remote_repo)
             log.info("PR #{} found.".format(existing_pr.number) if existing_pr else "PR not found.")
-            if existing_pr and not existing_pr.opened and not args.defaults and (args.defaults is False or Terminal.choose(
-                    "'{}' is already associated with '{}', which is closed.\nWould you like to create a new pull-request?".format(
-                        repository.branch, existing_pr,
-                    ), default='No',
+            if existing_pr and not existing_pr.opened and not args.defaults and (
+                args.defaults is False or Terminal.choose(
+                    "'{}' is already associated with '{}', which is closed.\nWould you like to create a new pull-request?".format(repository.branch, existing_pr),
+                    default='No',
             ) == 'Yes'):
                 existing_pr = None
 
+            if existing_pr and user and existing_pr.author != user and (args.defaults or Terminal.choose(
+                "'{}' is owned by '{}'\nWould you like to take over this pull-request?".format(existing_pr, existing_pr.author),
+                default='No',
+            ) == 'No'):
```

Comment:
FWIW, I tried this locally and saw what appear to be some line control glitches: 
<img width="1223" alt="Screen Shot 2022-09-07 at 16 57 33" src="https://user-images.githubusercontent.com/96081375/189005425-5fdd56f0-f840-452d-b2b0-a4b150b965f7.png">
Maybe because the prompt message doesn't end in a newline?

Reply:
Attempting a fix for this in #4131